controllers/util/util.go

```diff
@@ -103,3 +104,19 @@ func AssertEqual[T comparable](actual T, expected T, exitCode int) {
 		os.Exit(exitCode)
 	}
 }
+
+func IsClusterOrDeviceSetEncrypted(sc *ocsv1.StorageCluster) bool {
+	// If cluster-wide encryption is enabled
+	if sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide {
+		return true
+	}
+
+	// If any device set is encrypted
+	for _, deviceSet := range sc.Spec.StorageDeviceSets {
+		if deviceSet.Encrypted != nil && *deviceSet.Encrypted {
+			return true
```

Comment:
can we just `return *deviceSet.Encrypted` and do not check for &&

Reply:
That would be wrong as, If one deviceSet has false in the Encrypted field and the loop encounters that first it would return false from there & consequentially KMS details would not be added. But we expect to return true even if any one DeviceSet has Encrypted True so that KMS details are added.